hudi-sync/hudi-bigquery-sync/src/main/java/org/apache/hudi/bigquery/HoodieBigQueryClient.java

```diff
@@ -0,0 +1,215 @@
+              .setIgnoreUnknownValues(false)
+              .setMaxBadRecords(0)
+              .build();
+      bigquery.create(TableInfo.of(tableId, customTable));
+      LOG.info("Manifest External table created.");
+    } catch (BigQueryException e) {
+      throw new HoodieException("Manifest External table was not created ", e);
+    }
+  }
+
+  public void createSnapshotView(
+      String projectId, String datasetName, String viewName, String versionsTableName, String manifestTableName) {
+    try {
+      TableId tableId = TableId.of(projectId, datasetName, viewName);
+      String query =
+          String.format(
+              "SELECT * FROM `%s.%s.%s` WHERE _hoodie_file_name IN "
+                  + "(SELECT filename FROM `%s.%s.%s`)",
+              projectId,
+              datasetName,
+              versionsTableName,
+              projectId,
+              datasetName,
+              manifestTableName);
+
+      ViewDefinition viewDefinition =
+          ViewDefinition.newBuilder(query).setUseLegacySql(false).build();
+
+      bigquery.create(TableInfo.of(tableId, viewDefinition));
+      LOG.info("View created successfully");
+    } catch (BigQueryException e) {
+      throw new HoodieException("View was not created ", e);
+    }
+  }
+
+  @Override
+  public Map<String, String> getTableSchema(String tableName) {
+    // TODO: Implement automatic schema evolution when you add a new column.
+    return null;
+  }
+
+  @Override
+  public void addPartitionsToTable(final String tableName, final List<String> partitionsToAdd) {
+    // bigQuery discovers the new partitions automatically, so do nothing.
+  }
+
+  @Override
+  public void dropPartitionsToTable(final String tableName, final List<String> partitionsToDrop) {
+    // bigQuery discovers the new partitions automatically, so do nothing.
+  }
+
+  @Override
+  public boolean doesTableExist(final String tableName) {
+    // bigQuery table exists needs different set of arguments, so do nothing.
+    throw new UnsupportedOperationException("Not support doesTableExist yet.");
+  }
+
+  public boolean doesTableExist(final String projectId, final String datasetName, final String tableName) {
+    TableId tableId = TableId.of(projectId, datasetName, tableName);
+    return bigquery.getTable(tableId, BigQuery.TableOption.fields()) != null;
+  }
+
+  public boolean doesViewExist(final String projectId, final String datasetName, final String viewName) {
+    TableId tableId = TableId.of(projectId, datasetName, viewName);
+    return bigquery.getTable(tableId) != null;
+  }
+
+  @Override
+  public Option<String> getLastCommitTimeSynced(final String tableName) {
+    // bigQuery doesn't support tblproperties, so do nothing.
+    throw new UnsupportedOperationException("Not support getLastCommitTimeSynced yet.");
+  }
+
+  @Override
+  public void updateLastCommitTimeSynced(final String tableName) {
+    // bigQuery doesn't support tblproperties, so do nothing.
```

Comment:
can we standardize on all unsupported APIs to throw `UnsupportedOperationException` ?

Reply:
Good point, Resolved.